apidef/oas/authentication.go

```diff
@@ -229,71 +232,264 @@ func (a *Authentication) ExtractTo(api *apidef.APIDefinition) {
+}
+
+// Get retrieves a security scheme by name.
+// It returns the stored value and true when present, otherwise (nil, false).
+func (ss *SecuritySchemes) Get(key string) (interface{}, bool) {
+	if ss == nil {
+		return nil, false
+	}
+	ss.mutex.RLock()
+	defer ss.mutex.RUnlock()
+
+	if ss.container == nil {
+		return nil, false
+	}
+	value, ok := ss.container[key]
+	return value, ok
+}
+
+// Delete removes a scheme from the receiver by name.
+// It silently returns when the receiver or its container map is nil.
+func (ss *SecuritySchemes) Delete(key string) {
+	if ss == nil {
+		return
+	}
+	ss.mutex.Lock()
+	defer ss.mutex.Unlock()
+
+	if ss.container == nil {
+		return
+	}
+	delete(ss.container, key)
+}
+
+// Len reports the number of registered security schemes.
+// A nil receiver reports zero.
+func (ss *SecuritySchemes) Len() int {
+	if ss == nil {
+		return 0
+	}
+	ss.mutex.RLock()
+	defer ss.mutex.RUnlock()
+
+	return len(ss.container)
+}
+
+// Iter returns a snapshot under a short read lock to avoid holding locks during iteration.
+// This allocates per call. Given the typical small number of schemes, the trade-off is acceptable.
+func (ss *SecuritySchemes) Iter() iter.Seq2[string, interface{}] {
+	return func(yield func(string, interface{}) bool) {
+		if ss == nil {
+			return
+		}
+
+		ss.mutex.RLock()
+		if ss.container == nil {
+			ss.mutex.RUnlock()
+			return
+		}
+
+		snap := make([]struct {
+			k string
+			v interface{}
+		}, 0, len(ss.container))
+		for k, v := range ss.container {
+			snap = append(snap, struct {
+				k string
+				v interface{}
+			}{k, v})
+		}
+		ss.mutex.RUnlock()
+
+		for _, e := range snap {
+			if !yield(e.k, e.v) {
+				return
+			}
+		}
+	}
+}
+
+// MarshalJSON implements json.Marshaler.
+// It snapshots the map under RLock, unlocks, then marshals.
+func (ss *SecuritySchemes) MarshalJSON() ([]byte, error) {
+	if ss == nil {
+		return []byte(`{}`), nil
+	}
+
+	ss.mutex.RLock()
+	defer ss.mutex.RUnlock()
+
+	if ss.container == nil {
+		return []byte(`{}`), nil
+	}
+
+	return json.Marshal(ss.container)
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+// It builds a temporary map, then replaces the internal map under Lock.
+func (ss *SecuritySchemes) UnmarshalJSON(b []byte) error {
+	if string(b) == "null" {
+		ss.mutex.Lock()
+		ss.container = make(map[string]interface{})
+		ss.mutex.Unlock()
+		return nil
+	}
+
+	tmp := make(map[string]interface{})
+	if err := json.Unmarshal(b, &tmp); err != nil {
+		return err
+	}
+
+	ss.mutex.Lock()
+	ss.container = tmp
+	ss.mutex.Unlock()
+	return nil
+}
+
+// MarshalYAML makes SecuritySchemes YAML-friendly by exposing its container map.
+func (ss *SecuritySchemes) MarshalYAML() (interface{}, error) {
+	if ss == nil {
+		return map[string]interface{}{}, nil
+	}
+	ss.mutex.RLock()
+	defer ss.mutex.RUnlock()
+
+	if ss.container == nil {
+		return map[string]interface{}{}, nil
+	}
+	// Return the raw map so yaml can encode it normally.
+	return ss.container, nil
```

Comment:
This code does not like to be  safe.
Map resize can occur during yaml marshaling.
It would be better to return copy of the map.